### DISCUSSION TITLE
New release 1.6.4

### DISCUSSION THREAD

**Discussion by pablosnt** [pablosnt/rekono#259] [Announcements]
### Security

- Upgrade `Django` version to `3.2.23` (https://github.com/pablosnt/rekono/issues/252)

<hr /><em>This discussion was created from the release <a href='https://github.com/pablosnt/rekono/releases/tag/1.6.4'>1.6.4</a>.</em>

**Comment by ChristopfWW** [comment]
hello after i tried to install rekono desktop (apt install rekono-kbx)

i got a error on launch on kali linux can u plz help? :): 
```
# rekono                                                                                                                                               
rekono is at version 1.6.4, can't run rekono=1.6.3
NoneType: None
```

**Reply by pablosnt** [reply]
Okay, I've got the same problem. I will reach the Kali Linux team to see what is the problem. Thank you for reporting it, I hope to have a solution as soon as possible

**Reply by ChristopfWW** [reply]
thank you, maybe this will help...in the installation proces i can see this....

```
Setting up rekono-kbx (1.6.3) ...
Preparing rekono
Pulling registry.hub.docker.com/pablosnt/rekono:1.6.4 image from registry
Processing triggers for libc-bin (2.37-12) ...
Processing triggers for man-db (2.11.2-3) ...
Processing triggers for mailcap (3.70+nmu1) ...
Processing triggers for kali-menu (2023.4.5) ...
Processing triggers for desktop-file-utils (0.26-1) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
                                                                                                                                                                                               

└─# rekono
rekono is at version 1.6.4, can't run rekono=1.6.3
NoneType: None
                                                       
```

keep a good work im lookin forward to feedback from you :)

**Reply by pablosnt** [reply]
Thank you! I think this is going to be very useful

**Reply by pablosnt** [reply]
In the meantime, if you need to use rekono urgently, you can install it using the DEB file from the [1.6.4 release](https://github.com/pablosnt/rekono/releases/tag/1.6.4) while the official Kali Linux package is not fixed:

```
wget https://github.com/pablosnt/rekono/releases/download/1.6.4/rekono-kbx_1.6.4_amd64.deb && dpkg -i rekono-kbx_1.6.4_amd64.deb || apt -f install -y
```

It worked for me

**Reply by pablosnt** [reply]
The problem should be currently fixed in Kali Linux. Please, try again the installation @ChristopfWW.

I'm still working with Kali Linux's team to prevent this error in further releases. Thank you for your patience and my apologies for the inconvenience

**Comment by ChristopfWW** [comment]
hello @pablosnt thank you for your time, right now i got this errors, any help will be gr8, thank you in advance:

```
`root㉿kali)-[/home/kali]
└─# rekono             
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/docker/api/client.py", line 268, in _raise_for_status
    response.raise_for_status()
  File "/usr/lib/python3/dist-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http+docker://localhost/v1.41/containers/create

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/kaboxer", line 33, in <module>
    sys.exit(load_entry_point('kaboxer==1.1.4', 'console_scripts', 'kaboxer')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/kaboxer/__init__.py", line 2220, in main
    kaboxer.go()
  File "/usr/lib/python3/dist-packages/kaboxer/__init__.py", line 379, in go
    self.args.func()
  File "/usr/lib/python3/dist-packages/kaboxer/__init__.py", line 496, in cmd_run
    container = self.docker_conn.containers.create(image, **opts2)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/docker/models/containers.py", line 878, in create
    resp = self.client.api.create_container(**create_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/docker/api/container.py", line 428, in create_container
    return self.create_container_from_config(config, name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/docker/api/container.py", line 439, in create_container_from_config
    return self._result(res, True)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/docker/api/client.py", line 274, in _result
    self._raise_for_status(response)
  File "/usr/lib/python3/dist-packages/docker/api/client.py", line 270, in _raise_for_status
    raise create_api_error_from_http_exception(e)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/docker/errors.py", line 31, in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation)
docker.errors.ImageNotFound: 404 Client Error for http+docker://localhost/v1.41/containers/create: Not Found ("No such image: kaboxer/rekono:1.6.4")
`
```

**Reply by pablosnt** [reply]
You are welcome!

This error sometimes happens when the Docker environment is not clean (due to the previous installation attempts, for example). Please, check the Docker images that you have in your local machine. You are likely to have already pulled some rekono image in the previous installation attempts, if so, please remove the rekono images that you have, and install the `rekono-kbx` again. It's not very user-friendly, I know, but it should work. Let me know how is it going

**Comment by ChristopfWW** [comment]
@pablosnt thank you again for your time and feedback, i've installed all from a scratch and it works like a charm. I got 1 question if i may, can i somehow export results for example to .html when i scan project? Or the only option is to check the results in this gui?

**Reply by pablosnt** [reply]
You are welcome! Until now, the only options are the UI or the API Rest. You can make API Rest and generate the output that you need. Take a look at the [rekono-cli](https://github.com/pablosnt/rekono-cli), maybe it will be useful for you.

However, I know that Rekono needs more export options to handle the findings, so I'm already working on a `2.0.0` version, where I expect to add the export feature in the UI to generate reports in PDF or other formats like CSV, JSON, etc automatically. If you have any special need for that feature, let me know, as it will be developed soon and it's the perfect time to get ideas :)

**Comment by ChristopfWW** [comment]
> However, I know that Rekono needs more export options to handle the findings, so I'm already working on a 2.0.0 version, where I expect to add the export feature in the UI to generate reports in PDF or other formats like CSV, JSON, etc automatically. 

yes, export report with findings to pdf, html and such will be gr8 feature :) it will really helps to send some info to administrators :)
from my perspective: banners (version and tech/software/OS), SSL communication (some old TLS, weak ciphers etc.), vulnerabilities from findings in server banners for a start


you are doing the gr8 job @pablosnt - this is more than i've expected so far :) i'm feeling like i've just found a holy grail ;)

**Reply by pablosnt** [reply]
Thank you very much! ;)

**Comment by ChristopfWW** [comment]
@pablosnt can you also add some progress bar or some prompts from script console because its a little bit frustrating im wainting for an output from metasploit for 3h an nothing happens :) and scan is still runing :) and there is some issue with this scroll menu on the right i think :) 

thanks again! cheers!

![image](https://github.com/pablosnt/rekono/assets/150273917/f1c618ba-298c-4885-9fee-5c9dd2cf6794)

**Reply by ChristopfWW** [reply]
meantime can you tell how to monit those actions ?

**Reply by pablosnt** [reply]
I will explore the alternatives, but I'm not sure if it's possible to get the execution progress from Metasploit. Rekono just executes commands and then returns the output. Anyway, I will take a look at this issue because when I added support to Metasploit I saw that its execution through Python (and Rekono) takes a long time in comparison with the standard shell execution of the same command. So...I will take a look at the problem and see if we can improve this behavior. Thank you for reporting it!

In the meantime, I think there are not many options because I think Metasploit doesn't print anything during the execution of that command, so unfortunately you can't even check the Rekono logs...

**Comment by ChristopfWW** [comment]
@pablosnt sure thing, thank you again for your feedback and a quick response, im lookin forward for a hint, keep a good work ✌️
